exonum/src/runtime/rust/interfaces.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Important interservice communication interfaces.
+
+use crate::{
+    merkledb::{BinaryValue, Snapshot},
+    runtime::{CallContext, Caller, DispatcherError, MethodId, SUPERVISOR_SERVICE_ID},
+};
+
+use super::{ExecutionError, Interface, TransactionContext};
+
+/// Fully qualified name of the ['Configure`] interface.
+///
+/// ['Configure`]: trait.Configure.html
+pub const CONFIGURE_INTERFACE_NAME: &str = "Configure";
+/// Identifier of the [`Configure::verify_config`] method.
+///
+/// [`Configure::verify_config`]: trait.Configure.html#tymethod.verify_config
+pub const VERIFY_CONFIG_METHOD_ID: MethodId = 0;
```

Comment:
Can't method IDs be deduced automatically based on their order in the trait (like for service methods)?

Reply:
Better to have them explicitly defined, as other runtimes do not have information about order of methods.

Reply:
Then I think it'll be better to have a `Enum` or mapping "method_name" => "method_id" returned by service via `service.get_methods()` instead of constants.

Reply:
I don't like if the only way to access method IDs would be using some methods (because it has significant impact on performance in EJB), but I don't mind agains a public enum + optional methods for Rust

Reply:
Well, even if methods will be the only mechanism for retrieving such info, it's not the kind of data that you need to refresh often. If that's expensive, you always can call it only once per service/interface instance and cache it.

However, I don't like the idea of designing things in favor of some concrete target implementation. See dependency inversion principle to understand why it's bad.

Reply:
Yep, I do not suggest design runtimes in a way that is better for EJB. I just say that methods will involve additional complexity comparing to simple constants. Dealing with such complexity is possible (we know how to cache data), but not free in terms of time, code size and runtime performance.

Actually, do you need mapping "name => id"? Names are not used by user, so some kind of C-style enum would be more suitable here, wouldn't it?

Reply:
Yeah, I don't mind your suggestion, it was just a remark to be kept in mind.

Reply:
It can be deduced by the `exonum-derive` or certain codegen tool which whould be generate trait from the protobuf file spec. But it can't be deduced from the regular Rust code.